src/cpprealm/persisted.hpp

```diff
@@ -275,8 +302,251 @@ struct persisted_container_base : public persisted_base<T> {
     void push_back(const value_type& a) requires (type_info::PrimitivePersistable<value_type>);
     void push_back(value_type& a) requires (type_info::ObjectPersistable<value_type>);
     void push_back(value_type&& a) requires (type_info::ObjectPersistable<value_type>);
+
+    void pop_back();
+    void erase(size_type pos);
+    void clear();
+
+    void set(size_type pos, const value_type& a) requires (type_info::PrimitivePersistable<value_type>);
+    void set(size_type pos, value_type& a) requires (type_info::ObjectPersistable<value_type>);
+    void set(size_type pos, value_type&& a) requires (type_info::ObjectPersistable<value_type>);
+
+    size_t find(const value_type& a) requires (type_info::PrimitivePersistable<value_type>);
+    size_t find(const value_type& a) requires (type_info::ObjectPersistable<value_type>);
+    size_t find(value_type&& a) requires (type_info::ObjectPersistable<value_type>);
+
+    notification_token observe(util::UniqueFunction<void(persisted<T>&,
+                                                         CollectionChange,
+                                                         std::exception_ptr)>);
 };
 
+template <realm::type_info::ListPersistable T>
+void persisted_container_base<T>::push_back(const typename T::value_type& a) requires (type_info::PrimitivePersistable<typename T::value_type>) {
+    if (this->m_obj) {
+        auto lst = this->m_obj->template get_list<typename type_info::persisted_type<typename T::value_type>::type>(this->managed);
+        if constexpr (type_info::BinaryPersistable<typename T::value_type>) {
+            BinaryData core_type = BinaryData(reinterpret_cast<const char *>(a.data()), a.size());
+            lst.add(core_type);
+        } else {
+            auto as_core_type = static_cast<typename type_info::persisted_type<typename T::value_type>::type>(a);
+            lst.add(as_core_type);
+        }
+    } else {
+        this->unmanaged.push_back(a);
+    }
+}
+
+template <realm::type_info::ListPersistable T>
+void persisted_container_base<T>::push_back(typename T::value_type& a)
+requires (type_info::ObjectPersistable<typename T::value_type>) {
+    if (this->m_obj) {
+        auto lst = this->m_obj->template get_list<typename type_info::persisted_type<typename T::value_type>::type>(this->managed);
+        if (!a.m_obj) {
+            T::value_type::schema::add(a, this->m_obj->get_table()->get_link_target(this->managed), nullptr);
+        }
+        lst.add(a.m_obj->get_key());
+    } else {
+        this->unmanaged.push_back(a);
+    }
+}
+
+template <realm::type_info::ListPersistable T>
+void persisted_container_base<T>::push_back(typename T::value_type&& a)
+requires (type_info::ObjectPersistable<typename T::value_type>) {
+    push_back(a);
```

Comment:
Why isn't this a `std::move`? This will behave differently than how vectors are meant to behave here.

Reply:
oversight